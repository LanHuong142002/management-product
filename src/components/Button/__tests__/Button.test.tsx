import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import { Button, ButtonProps } from '@components';

describe('Testing button component', () => {
  afterEach(() => {
    cleanup();
  });

  const defaultProps = {
    label: 'Cancel',
    variant: 'primary',
    size: 'lg',
  } as ButtonProps;

  it('Should render Button', () => {
    const { container } = render(<Button {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });

  it('Should call click event when click button', () => {
    const handleClick = jest.fn();
    render(<Button {...defaultProps} onClick={handleClick} />);

    const button = screen.getByRole('button');
    fireEvent.click(button);

    expect(handleClick).toBeCalled();
    expect(handleClick).toBeCalledTimes(1);
  });

  it('Should render default button correctly', () => {
    render(<Button label='Cancel' />);

    const button = screen.getByRole('button');

    expect(button).toBeInTheDocument();
  });

  it('Should render button with disabled correctly', () => {
    const handleClick = jest.fn();
    render(<Button label='Cancel' isDisabled={true} onClick={handleClick} />);

    const button = screen.getByRole('button');
    fireEvent.click(button);

    expect(handleClick).not.toBeCalled();
  });

  it('Should render button with loading correctly', () => {
    const handleClick = jest.fn();
    render(<Button label='Loading' isLoading={true} onClick={handleClick} />);

    const button = screen.getByRole('button');
    fireEvent.click(button);

    expect(button).toHaveAttribute('disabled');
    expect(handleClick).not.toBeCalled();
  });
});
