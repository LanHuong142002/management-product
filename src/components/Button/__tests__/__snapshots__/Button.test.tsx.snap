// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing button component Should render Button 1`] = `
<div>
  <button
    class="btn btn-primary btn-color-default btn-lg  "
    type="button"
  >
    <span>
      Cancel
    </span>
  </button>
</div>
`;
